app/controllers/incidents_controller.rb

```diff
@@ -49,6 +49,14 @@ def edit
     end
   end
 
+  def batch_export
+    @incidents = Incident.where(id: params[:ids])
+    @incidents.each(&:mark_as_exported)
```

Comment:
I would mark them as exported _after_ `send_data`. Just bothers me to say "I did the thing!" right before doing the thing.

Reply:
I suppose that's true, that if for some reason the file doesn't render, then we shouldn't mark it as exported. I guess I'm just so used to not putting code after a render in a controller action.